src/renderer/Renderer.js

```diff
@@ -184,17 +188,20 @@ export default class Renderer {
                     continue;
                 }
 
-                viewportExpressions.forEach(viewportExpression => viewportExpression.accumViewportAgg(feature));
+                for (let j = 0; j < viewportExpressionsLength; j++) {
+                    viewportExpressions[j].accumViewportAgg(feature);
+                }
             }
         });
     }
 
     /**
      * Check if the feature at the "index" position of the given dataframe is in the renderer viewport.
+     * NOTE: requires `this.aspect` to be set
      */
     _isFeatureInViewport(dataframe, index) {
```

Comment:
what about passing the aspect as a parameter?

Reply:
ok...